### ISSUE TITLE
Triggering the calculation of liabilities

### ISSUE THREAD

**Issue by andyws** [hmrc/self-assessment-api#262]
A while ago I asked the question wrt SA API: "Will updates be possible at any time or just once a quarter with a final set of adjustments at the year end?"

I received this reply from SDS:

> With regards to your question "Will updates be possible at any time or just once a quarter with a final set of adjustments at the year end?", customers will be able to provide an update at any time, although we are likely to limit this to no more than once a day.

I have reflected on this answer and I’m puzzled.  

Does your answer mean that if the SA api is utilized for a specific UTR several times in one day that:

- Liabilities will not be calculated until COB on that day?, 
or 
- Liabilities will be calculated only on the first data change? 
or
- Only one set of updates be allowed per UTR per day?  

As there is nothing in the API to allow me to trigger a [re]calculation other than changing data or aggregating discrete updates I am rather puzzled…

**Comment by saurabharora80** [comment]
Hi Andy,

Thanks for your question. 
Let me explain this in simpler terms. 
Our APIs are divided in 2 broad categories: Data Submission (e.g Business incomes and expenses) and Liability Calculation. 
There is no restriction on usage of Data submission APIs but Liability Calculation is restricted to at most once per day. 
i.e you can submit annual and quarterly data for a single business and a uk property (as per MVP scope) and then request a Liability calculation for the submitted data. You may update same or some other data on the same day, but you cant request another Liability calculation, as Liability calculation is restricted to at most once per day. 

It's important to understand that Liability is not automatically calculated when data is submitted, it must be request via a separate resource i.e POST /self-assessment/ni/{nino}/liability. 

This is a MPV restriction, which may be lifted as a later day. 

I hope I have answered you question. Please revert if you still have some questions.

**Comment by andyws** [comment]
Thanks for spelling this out.  I had not correctly understood that a POST to  /self-assessment/ni/{nino}/liability is the trigger for a liability calc.  

Is there any way of determining from the api whether the liabilities returned from the api include all pending items given that a calc may or may not have been triggered earlier in the day?


have you guys noticed that SA has disappeared from the dev hub?

**Comment by rakeshavasarala** [comment]
Hi Andy,

The SA API documentation has been restricted to Private Beta Dev's as it is still in Beta phase. You need to be logged in to view the documentation in Dev Hub.

Regards
Rakesh

**Comment by saurabharora80** [comment]
Hi Andy, 

A liability calculation is a year to day value and we shall consider all the the previously submitted data (for a given financial year) to compute the year's liability. 

I hope this answers your question.

regards
saurabh

**Comment by andyws** [comment]
@rakeshavasarala 

Thanks

**Comment by andyws** [comment]
@saurabharora80 

I had in mind the case that the HMRC api for a given utr/nino might be fed from multiple independent streams.  Without being able to determine if a calc has already been triggered today I will not be able to present the liability calculation retrieved by my client as reflecting the inputs s/he has just entered using my software until tomoz...

**Comment by saurabharora80** [comment]
Thanks. That is an interesting use case. 
Given the way things stand right now (MVP), the second liability call will fail. Going forward we may be able to allow one liability calculation per day per package; but that would have to be a business decision. So I cant make any promises or provide a timeline for this future.

**Comment by andyws** [comment]
@saurabharora80 

Thanks.  It is important to tease out such cases so as to avoid irritated customers.. ;-)

**Comment by saurabharora80** [comment]
@andyws  Absolutely. ;)

**Comment by saurabharora80** [comment]
@andyws Hi Andy, Is it alright if I close this issue?
